Homeworks/homework_6/despetcher_system.py

```diff
@@ -0,0 +1,87 @@
+import csv
+from itertools import count
+from datetime import datetime
+from random import sample, choice
+from consts import AIRCRAFT_TYPES
+from airport_factory import AirportFactory
+from aircraft_producing_company import AircraftProducingCompany
+
+counter = count(start=1)
+
+
+class DispatcherSystem:
+    __instance = None
+
+    def __new__(cls):
+        if cls.__instance is None:
+            cls.__instance = super().__new__(cls)
+        return cls.__instance
+
+    def __init__(self):
+        self.current_time = datetime.now()
+        self.airports = self.initial_setup
+
+    @property
+    def initial_setup(self):
+        with open('airports_.csv', 'r', encoding='utf-8') as csv_file:
+            chosen_airports = sample(list(csv.DictReader(csv_file)), 20)
+            airport_factory = AirportFactory()
+            airports = [airport_factory.create_airport(info) for info in chosen_airports if info is not None]
+        aircraft_factory = AircraftProducingCompany()
+        aircrafts_amount = 0
+        while aircrafts_amount < 20:
+            aircraft = aircraft_factory.produce_aircraft(choice(AIRCRAFT_TYPES), next(counter))
+            airport = choice(airports)
+            if airport.check_if_aircraft_can_land(aircraft):
+                airport.hangars.append(aircraft)
+                aircrafts_amount += 1
+        return airports
+
+    def describe_aircraft(self, aircraft_id):
+        for port in self.airports:
+            for aircraft in port.hangars:
+                if aircraft.id == aircraft_id:
+                    print(aircraft.__dict__)
```

Comment:
look, you overloaded __str__ method, so try to print just aircraft object but not a __dict__